### PR TITLE
Use the -mno-outline-atomics flag only for ARM64 targets

### PR DESCRIPTION
**Pull request by jason-simmons**

### CODE REVIEW

File: build/config/compiler/BUILD.gn [hunk 1]
```diff
@@ -416,13 +416,15 @@ config("compiler") {
...
       "-fno-short-enums",
       "-nostdinc++",
-      # TODO (https://github.com/flutter/flutter/issues/75348).
-      "-mno-outline-atomics",
     ]
     if (!is_clang) {
       # Clang doesn't support these flags.
       cflags += [ "-finline-limit=64" ]
     }
+    if (current_cpu == "arm64") {
```
Comment (iskakaushik) [inline build/config/compiler/BUILD.gn:424]: shoudln't this be `target_cpu`?

Comment (jason-simmons) [reply build/config/compiler/BUILD.gn:424]: `current_cpu` is generally preferred in build rules and will be equivalent to `target_cpu` in this case

See https://chromium.googlesource.com/experimental/chromium/src/+/refs/wip/bajones/webvr/tools/gn/docs/cross_compiles.md#as-a-build_gn-author

Comment (chinmaygarde) [reply build/config/compiler/BUILD.gn:424]: `target_cpu` will also hold on host builds. `current_cpu` is accurate.

**Review by chinmaygarde — APPROVED** [review]

File: build/config/compiler/BUILD.gn (hunk 1 above)
Comment (iskakaushik) [reply build/config/compiler/BUILD.gn:424]: Ah, good to know :-) LGTM!

**Pull request merged**